plugins/inputs/snmp/snmp.go

```diff
@@ -535,8 +538,8 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 		Time: time.Now(), //TODO record time at start
 		Rows: make([]RTableRow, 0, len(rows)),
 	}
-	for _, r := range rows {
-		if len(r.Tags) < tagCount {
+	for i, r := range rows {
+		if len(r.Tags)+emptyTags[i] < tagCount {
 			// don't add rows which are missing tags, as without tags you can't filter
 			continue
 		}
```

Comment:
More or less. The `emptyTags` thing basically allows inserting measurements where the plugin was configured with a tag and the tag ended up being empty, but blocks measurements where there was no tag configured at all.
If we just allow tagless measurements period, then the `emptyTags` stuff goes away yes.